src/js/requestHandlers.js

```diff
@@ -230,6 +235,83 @@ sjrk.storyTelling.server.deleteStoryFromCouch = function (id, deleteStoryDataSou
     return promise;
 };
 
+sjrk.storyTelling.server.deleteStoryFiles = function (storyContent, serverConfig) {
+    var filesToDelete = [];
+
+    fluid.each(storyContent, function (block) {
+        var blockFileName = "";
+
+        if (block.blockType === "image") {
+            blockFileName = block.imageUrl;
+        } else if (block.blockType === "audio" || block.blockType === "video") {
+            blockFileName = block.mediaUrl;
+        }
+
+        if (sjrk.storyTelling.server.isValidMediaFilename(blockFileName)) {
+            filesToDelete.push(blockFileName);
+        }
+    });
+
+    // remove duplicate entries so we don't try to delete already-deleted files
+    filesToDelete = filesToDelete.filter(function (fileName, index, self) {
+        return self.indexOf(fileName) === index;
+    });
+
+    fluid.each(filesToDelete, function (fileToDelete) {
+        sjrk.storyTelling.server.deleteSingleFileRecoverable(fileToDelete, serverConfig);
+    });
+};
+
+/*
+ * Verifies that a given file name follows the UUID format as laid out in
+ * RFC4122. A detailed description of the format can be found here:
+ * https://en.wikipedia.org/wiki/Universally_unique_identifier#Format
+ */
+sjrk.storyTelling.server.isValidMediaFilename = function (fileName) {
+    if (fileName && typeof fileName === "string" ) {
+        var fileNameWithoutExtension = path.basename(fileName, path.extname(fileName));
+
+        var validUuid = /\b[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\b/;
```

Comment:
Lets test the whole name with `$` and `^` and remove the use of `\b` -- let's test the whole filename, rather than testing a substring.